x-pack/plugins/reporting/server/native_controller_client/native_controller_channel.js

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+
+import uuid from 'uuid';
+
+export class NativeControllerChannel {
+  _messageCallbacks = [];
+  _messageAcks = {};
+
+  constructor(nativeControllerProcess, processId) {
+    this._process = nativeControllerProcess;
+    this._processId = processId;
+
+    this._process.on('message', ({ processId, ack, message }) => {
+      if (this._processId !== processId) {
+        return;
+      }
+
+      if (ack) {
+        const messageAck = this._messageAcks[ack.id];
+        if (ack.success) {
+          messageAck.resolve();
+        } else {
+          messageAck.reject(ack.error);
+        }
+        return;
```

Comment:
This means that any message with an ack will never be passed along to any registered callbacks. Shouldn't it resolve the promises, but also call any registered callbacks? Or am I missing something obvious?

Reply:
We're only including a `message` or an `ack`. The messages are for events that the process itself is responding with, while the acks are just for resolving callbacks.